test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2529,8 +2529,7 @@ public static <T> T safeGet(Future<T> future) {
      *
      * @return The value with which the {@code future} was completed.
      */
-    // NB private because tests should be designed not to need to wait for longer than SAFE_AWAIT_TIMEOUT.
-    private static <T> T safeGet(Future<T> future, TimeValue timeout) {
+    public static <T> T safeGet(Future<T> future, TimeValue timeout) {
```

Comment:
I don't think we can simply require tests to not have to wait for more than 10 seconds. In this case, downsampling is not consistently finished within 10 seconds, we really need more time. I'm open to other suggestions, but I think it's acceptable to allow consumers to specify a higher timeout.

Reply:
I'd rather we didn't make it so easy to abuse these utilities like this - experience shows that it just leads to lazy testing that waits for far too long and slows everything down. If it's just for a health request, you can pick your own timeout and use `ElasticsearchAssertions#assertNoTimeout`.

Relatedly, why is downsampling so slow in tests? Are we just sending it unreasonably large amounts of data, or is it slow for some other reason?

Reply:
Besides that, I feel like the difference between `TEST_REQUEST_TIMEOUT` (30s) and `SAFE_AWAIT_TIMEOUT` (10s) is also confusing. If we truly believe tests shouldn't have to wait for more than 10 seconds, we should drop the `TEST_REQUEST_TIMEOUT` too - I don't think we should, but that would be consistent.

Reply:
@DaveCTurner any more thoughts on this? Otherwise I'll ask someone from Data Management to review as well to get this through.

Reply:
I switched to `assertNoTimeout` in [c2a30b0](https://github.com/elastic/elasticsearch/pull/126773/commits/c2a30b0add7114f73087acc81e749bd0233c2179).